oat-sa/extension-tao-testqti-previewer
# Fix/TR 2892/resolve cloudfront assets

Related to https://oat-sa.atlassian.net/browse/TR-2892

Backend part: check url and credentials here: https://oat-sa.atlassian.net/browse/TR-2854?focusedCommentId=160358
With CDN, asset 'key' and 'value' are different: https://oat-sa.atlassian.net/browse/TR-2892?focusedCommentId=160370
This means `"packedUrl"` `assetManager` strategy should be used to resolve it: https://github.com/oat-sa/tao-item-runner-fe/blob/master/src/assets/strategies.js#L58
For that strategy to work we need to pass `itemData.assets` to `assetManager`. It's already done by 'normal' test runner: https://github.com/oat-sa/tao-test-runner-qti-fe/blob/master/src/provider/qti.js#L705 . So backend fix already works in 'normal' delivery launch.
But we also need it in 'previewer' test runner from this repo.

How to test:
- imitate CDN backend: e.g. in previewer proxy https://github.com/oat-sa/extension-tao-testqti-previewer/blob/master/views/js/previewer/proxy/item.js#L182 :
```
getItem: function getItem(itemIdentifier, params) {
	return this.request(
		....
	).then(itemData => itemDataHandlers(itemData)).then((itemData) => {
	        //your absolute url - copy from test run in delivery and add some param in the end
		const a = "https://tao.docker.localhost/tao/getFileFlysystem.php/618bc2b4498ca/1637334468/2b74874c69dac3aa82e44401ace5e50d/3/8/1/a5b13dbed706e0f92e8e4819266b0/*/en-US/61951e39478e5_eb495fa75d2ee93d5c3ec.jpg?aaaa=bbb";
		//your asset key
		itemData.content.assets.img["61951e39478e5_eb495fa75d2ee93d5c3ec.jpg"] = a;
		return itemData;
	});
}
``` 
(or better run preview on real env with CDN and backend fix: https://oat-sa.atlassian.net/browse/TR-2854?focusedCommentId=160358 , and copy real url and itemData from `getItem` response there. Test example: _Assembled Delivery -> QA -> Delivery of Test 3 -> first test taker_)
- run preview: 
![Screenshot_2](https://user-images.githubusercontent.com/38751932/142903672-b599ed73-1cb6-4521-9f17-19b626e2100c.png)
- check that images in preview have full url you've set in the first step 
- run unit tests
